xbmc/addons/addoninfo/AddonInfo.cpp

```diff
@@ -111,7 +117,7 @@ TYPE CAddonInfo::TranslateType(const std::string& string)
 {
   for (const TypeMapping& map : types)
   {
-    if (string == map.name || (!map.old_name.empty() && string == map.old_name))
+    if (string == map.name || (strlen(map.old_name) > 0 && string == map.old_name))
```

Comment:
Not sure. Does the comparison still work, now that `map.old_name` and `map.name` are `const char*`?

Reply:
About here I hope that with C++17 another way works and hopefully it can be `std::string_view` in `constexpr` and works with string compare.